File: 3way/problem2/partitionString.js
/**
 * @param {string} s
 * @return {number}
 */
var partitionString = function(s) {
    let seen = new Array(26).fill(false);
    let result = 1;

    for(let i = 0; i < s.length; i++){
        let index = s.charCodeAt(i) - 'a'.charCodeAt(0);;
        if(seen[index]){
            result++;
            seen = new Array(26).fill(false);
        }
        seen[index] = true
    }
    return result;
};

File: 3way/problem2/partitionString.c
int partitionString(char* s) {
    bool seen[26] = {false};
    int result = 1;

    for(int i = 0; s[i] != '\0'; i++){
        int index = s[i] - 'a';
        if(seen[index]){
            result++;
            memset(seen, 0, sizeof(seen));
        }
        seen[index] = true;
    }

    return result;
    
}
